src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -138,6 +138,13 @@ protected void AddVariablesToEnvironment(bool excludeNames = false, bool exclude
                 string formattedKey = (pair.Key ?? string.Empty).Replace('.', '_').ToUpperInvariant();
                 AddEnvironmentVariable(formattedKey, pair.Value);
 
+                // for maintaining back compat variable names replace space with underscore
+                string backCompatKey = formattedKey.Replace(' ', '_');
```

Comment:
i'm not sure we need two variables. there is a disconnect in the task lib (only replaces `.`), and we never heard complaints.

it sounds like a really rare occurrence that someone would have a variable name with a space in it. it's almost asking for trouble :)